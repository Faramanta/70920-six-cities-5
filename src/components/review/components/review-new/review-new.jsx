import {withReviewNew} from "../../../../hocs/with-review-new";
import Rating from "@components/rating/rating";

const ReviewNew = (props) => {
  const {rating, onSubmit, onChange} = props;

  return (
    <form
      className="reviews__form form"
      action="#"
      method="post"
      onSubmit={onSubmit}
    >
      <label className="reviews__label form__label" htmlFor="review">Your review</label>
      <Rating onChange={onChange} rating={rating} />
      <textarea
        className="reviews__textarea form__textarea"
        id="review"
        name="review"
        placeholder="Tell how was your stay, what you like and what can be improved"
        onChange={onChange}
      ></textarea>
      <div className="reviews__button-wrapper">
        <p className="reviews__help">
          To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
        </p>
        <button className="reviews__submit form__submit button" type="submit" disabled="">Submit</button>
      </div>
    </form>
  );
};


ReviewNew.propTypes = {
  rating: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default withReviewNew(ReviewNew);
