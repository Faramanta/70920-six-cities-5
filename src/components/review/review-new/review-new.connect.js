import ReviewNew from "./review-new";
import {connect} from "react-redux";
import {sendNewComment} from "@store/api-actions";

const mapDispatchToProps = (dispatch) => ({
  onSubmitAction: (offerId, commentData) => dispatch(sendNewComment(offerId, commentData)),
});

export default connect(null, mapDispatchToProps)(ReviewNew);
