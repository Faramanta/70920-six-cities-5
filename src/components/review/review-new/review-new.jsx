import {Fragment} from 'react';
import {ratingInfo, ReviewLimits} from "@const";
import {extend} from "@utils/utils";

const ReviewNew = (props) => {
  const {onSubmitAction, idCurrentOffer} = props;

  const [isFormStatus, setFormStatus] = React.useState(false);
  const [isError, setError] = React.useState(false);

  const [comment, setComment] = React.useState({
    rating: ``,
    review: ``,
  });
  const {rating, review} = comment;

  React.useEffect(() => {
    setError(false);
  }, [comment]);

  const handleFormSubmit = (evt) => {
    evt.preventDefault();
    if (rating && review) {
      setFormStatus(true);
      onSubmitAction(idCurrentOffer, {rating, review})
        .then(() => {
          setComment({
            rating: ``,
            review: ``,
          });
          setFormStatus(false);
        })
        .catch(() => {
          setError(true);
          setFormStatus(false);
        });
    }
  };

  const handleFieldChange = (evt) => {
    const {name, value} = evt.target;
    setComment((prevComment) => extend(prevComment, {[name]: value}));
  };

  const isButtonDisabled = () => {
    return !rating || review.length < ReviewLimits.MIN_COUNT_SIMBOLS || review.length > ReviewLimits.MAX_COUNT_SIMBOLS;
  };

  return (
    <form
      className="reviews__form form"
      action="#"
      method="post"
      onSubmit={handleFormSubmit}
    >
      <label className="reviews__label form__label" htmlFor="review">Your review</label>

      <div className="reviews__rating-form form__rating" >
        {ratingInfo.map((ratingItem) => {
          return (
            <Fragment key={`${ratingItem.value}`} >
              <input
                className="form__rating-input visually-hidden"
                name="rating"
                value={ratingItem.value}
                id={`${ratingItem.value}-stars`}
                type="radio"
                onChange={(evt) => handleFieldChange(evt)}
                checked={rating === ratingItem.value}
                disabled={isFormStatus}
              />
              <label htmlFor={`${ratingItem.value}-stars`} className="reviews__rating-label form__rating-label" title={ratingItem.title}>
                <svg className="form__star-image" width="37" height="33">
                  <use xlinkHref="#icon-star"></use>
                </svg>
              </label>
            </Fragment>
          );
        })}
      </div>
      <textarea
        className="reviews__textarea form__textarea"
        id="review"
        name="review"
        placeholder="Tell how was your stay, what you like and what can be improved"
        onChange={(evt) => handleFieldChange(evt)}
        value={review}
        disabled={isFormStatus}
      ></textarea>
      <div className="reviews__button-wrapper">
        <p className="reviews__help">
          To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
        </p>
        <button
          className="reviews__submit form__submit button" type="submit"
          disabled={isButtonDisabled()} style={isError ? {background: `red`} : {background: ``}}>{isError ? `Sending error` : `Submit`}</button>
      </div>
    </form>
  );
};

ReviewNew.propTypes = {
  idCurrentOffer: PropTypes.number.isRequired,
  onSubmitAction: PropTypes.func,
};

export default ReviewNew;
