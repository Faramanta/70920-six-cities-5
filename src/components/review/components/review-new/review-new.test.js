import renderer from 'react-test-renderer';
import {ReviewNew} from "./review-new";

describe(`ReviewNew render correctly`, () => {
  it(`ReviewNew all fields are filled`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={`What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river`}
            rating={`4`}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewNew rating field empty`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={`What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river`}
            rating={``}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewNew review field empty`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={``}
            rating={`4`}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewNew all field empty`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={``}
            rating={``}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewNew very short review`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={`very short text`}
            rating={`4`}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`ReviewNew very long review`, () => {
    const tree = renderer
      .create(
          <ReviewNew
            idCurrentOffer={4}
            review={`Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum`}
            rating={`4`}
            onSubmitAction={() => {}}
            onChange={() => {}}
            changeDisabledFormStatus={() => {}}
            resetForm={() => {}}
            isDisabled={false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
