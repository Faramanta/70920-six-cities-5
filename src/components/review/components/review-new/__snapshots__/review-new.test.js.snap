// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewNew render correctly ReviewNew all field empty 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;

exports[`ReviewNew render correctly ReviewNew all fields are filled 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={true}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value="What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river"
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;

exports[`ReviewNew render correctly ReviewNew rating field empty 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value="What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river"
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;

exports[`ReviewNew render correctly ReviewNew review field empty 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={true}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;

exports[`ReviewNew render correctly ReviewNew very long review 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={true}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;

exports[`ReviewNew render correctly ReviewNew very short review 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={true}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      disabled={false}
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    disabled={false}
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value="very short text"
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "16",
      }
    }
  />
</form>
`;
